programs/libreplex_editions_controls/src/instructions/mint_with_controls.rs

```diff
@@ -49,130 +37,246 @@ pub struct MintWithControlsCtx<'info> {
+    pub group_mint: UncheckedAccount<'info>,
+
+    /// CHECK: Platform fee recipient
+    #[account(mut)]
+    pub platform_fee_recipient_1: UncheckedAccount<'info>,
+
+    // TODO add optimization for stack
+    // #[account(mut)]
+    // pub platform_fee_recipient_2: UncheckedAccount<'info>,
+
+    /// CHECK: Passed in via CPI to mpl_token_metadata program
     #[account(mut)]
     pub token_account: UncheckedAccount<'info>,
-    
+
     /// CHECK: Checked in constraint
-    #[account(mut,
-        constraint = editions_controls.treasury == treasury.key())]
+    #[account(
+        mut,
+        constraint = editions_controls.treasury == treasury.key()
+    )]
     pub treasury: UncheckedAccount<'info>,
 
-
-    /* BOILERPLATE PROGRAM ACCOUNTS */
     /// CHECK: Checked in constraint
     #[account(
-        constraint = token_program.key() == token_2022::ID
+        constraint = token_program.key() == TOKEN_2022_ID
     )]
     pub token_program: UncheckedAccount<'info>,
 
-    #[account()]
     pub associated_token_program: Program<'info, AssociatedToken>,
 
-    /// CHECK: address checked
+    /// CHECK: Address checked
     #[account(address = group_extension_program::ID)]
-    pub group_extension_program: AccountInfo<'info>,
+    pub group_extension_program: UncheckedAccount<'info>,
 
-    #[account()]
     pub system_program: Program<'info, System>,
 
-    pub libreplex_editions_program: Program<'info, LibreplexEditions>
-
+    pub libreplex_editions_program: Program<'info, LibreplexEditions>,
 }
 
-
-
-pub fn mint_with_controls(ctx: Context<MintWithControlsCtx>, mint_input: MintInput) -> Result<()> {
-    
-    let libreplex_editions_program = &ctx.accounts.libreplex_editions_program;
-    let editions_deployment = &ctx.accounts.editions_deployment;
+pub fn mint_with_controls(
+    ctx: Context<MintWithControlsCtx>,
+    mint_input: MintInput,
+) -> Result<()> {
     let editions_controls = &mut ctx.accounts.editions_controls;
-   
-    let hashlist = &ctx.accounts.hashlist;
-    let hashlist_marker = &ctx.accounts.hashlist_marker;
-    let payer = &ctx.accounts.payer;
-    let mint = &ctx.accounts.mint;
-    let token_account = &ctx.accounts.token_account;
-    let associated_token_program = &ctx.accounts.associated_token_program;
-    let minter = &ctx.accounts.minter;
-    let group = &ctx.accounts.group;
-    let system_program = &ctx.accounts.system_program;
-    let token_program = &ctx.accounts.token_program;
     let minter_stats = &mut ctx.accounts.minter_stats;
-    let treasury = &ctx.accounts.treasury;
     let minter_stats_phase = &mut ctx.accounts.minter_stats_phase;
-    let group_extension_program = &ctx.accounts.group_extension_program;
-    let member = &ctx.accounts.member;
-    if mint_input.phase_index >= editions_controls.phases.len() as u32 {
+    let minter = &ctx.accounts.minter;
+
+    // Phase validation and price retrieval
+    let price_amount = validate_phase(editions_controls, mint_input.phase_index)?;
```

Comment:
I suggest letting this function be in charge of only its name purpose, validating the phase, not necessarily also determining and returning the phase price, this can be taken in a one-liner below, to let validate_phase be more concise and reusable